Readability.js

```diff
@@ -1292,12 +1310,81 @@ Readability.prototype = {
     });
   },
 
+  /**
+   * Try to extract metadata from JSON-LD object.
+   * For now, only Schema.org objects of type Article or its subtypes are supported.
+   * @return Object with any metadata that could be extracted (possibly none)
+   */
+  _getJSONLD: function (doc) {
+
+    var scripts = this._getAllNodesWithTag(doc, ["script"]);
+
+    var jsonLdElement = this._findNode(scripts, function(el) {
+      return el.getAttribute("type") === "application/ld+json";
+    });
+
+    if (jsonLdElement) {
+      try {
+        // Strip CDATA markers if present
+        var content = jsonLdElement.textContent.replace(/^\s*<!\[CDATA\[\s*|\s*\]\]>\s*$/g, "");
```

Comment:
This replaces whitespace around the CDATA bits as well. Is that necessary? I would have assumed that the JSON parsing would take care of that - or are there CDATA things *inside* the JSON? _*shudder*_

Reply:
Oops, good catch! It's only the trailing whitespace that needs specifying.